### PR TITLE
dns: add ability to get CNAME with evdns_base_resolve_ipv4()/evdns_base_resolve_ipv6() 

### PR DESCRIPTION
**Pull request by sem-hub**
dns: add ability to get CNAME with evdns_base_resolve_ipv4()/evdns_base_resolve_ipv6() by setting
DNS_CNAME_CALLBACK flag. After that we get one more callback with type == DNS_CNAME and CNAME in addrs argument.

### CODE REVIEW

File: evdns.c [hunk 1]
```diff
@@ -986,12 +988,15 @@ reply_run_callback(struct event_callback *d, void *user_pointer)
...
 	case TYPE_A:
-		if (cb->have_reply)
+		if (cb->have_reply) {
 			cb->user_callback(DNS_ERR_NONE, DNS_IPv4_A,
 			    cb->reply.rr_count, cb->ttl,
 			    cb->reply.data.a,
 			    user_pointer);
-		else
+			if (cb->reply.cname)
+			    cb->user_callback(DNS_ERR_NONE, DNS_CNAME, 1,
```
Comment (azat) [inline evdns.c:997]: Minor - tabulation looks wrong (and please check the whole patch)

**Review by azat — COMMENTED** [review]
Looks OK, can you please also add a unit test in test/regress_dns.c?

File: evdns.c (hunk 1 above)
Comment (azat) [reply evdns.c]: >BTW ./checkpatch.sh -r as recomended in CONTRIBUTING.md changes too much exist code. So I skipped it.

Yes, you did everything right, thank you!

**Comment by azat** [comment]
@sem-hub can you also add a test into regress_dns.c?

**Comment by sem-hub** [comment]
> @sem-hub can you also add a test into regress_dns.c?

OK. I will do.

**Review by azat — APPROVED** [review]

**Pull request merged**